<launch>
  
  
  <arg name="fake_state" default="false"/>
  
  <!-- send robot urdf to param server -->
  <param name="robot_description" command="$(find xacro)/xacro.py $(find nao_description)/urdf/nao_robot_v4.urdf.xacro" />

  <!-- send fake joint values -->
  <group if="$(arg fake_state)">
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
      <param name="use_gui" value="true" />
    </node>
  </group>
  
  <!-- PrioVR node -->
  <group unless="$(arg fake_state)">
    <node name="get_joint_states" pkg="priovr" type="get_joint_states.py" output="screen">
      <rosparam command="load" file="$(find priovr)/config/priovr_common.yaml"/>
      <rosparam command="load" file="$(find priovr)/config/nao.yaml"/>
    </node>
  </group>

  <!-- Combine joint values -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />

  <!-- Show in Rviz  -->
  <node name="rviz" pkg="rviz" type="rviz" output="screen"/> 
</launch>
